torch/_lobpcg.py

```diff
@@ -11,7 +11,92 @@
 from .overrides import has_torch_function, handle_torch_function
 
 
-__all__ = ['lobpcg']
+__all__ = ['lobpcg', 'lobpcg2']
+
+class LOBPCGAutogradFunction(torch.autograd.Function):
```

Comment:
Once `gradcheck` passes, we will need to come up with a proper name for the generalized eigenvalue function. It makes sense to probably fix some parameters of `lobpcg` to make the forward lighter.
cc @pearu